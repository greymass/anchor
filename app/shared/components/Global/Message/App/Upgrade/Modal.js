// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';

import { Modal } from 'semantic-ui-react';

import GlobalMessageAppUpgrade from '../Upgrade';

export class GlobalMessageAppUpgradeModal extends Component<Props> {
  render() {
    const {
      actions,
      constants,
      name,
      version
    } = this.props;
    return (
      <Modal
        content={(
          <GlobalMessageAppUpgrade
            actions={actions}
            constants={constants}
            name={name}
            version={version}
          />
        )}
        open
        size="small"
      />
    );
  }
}

export default translate('global')(GlobalMessageAppUpgradeModal);
