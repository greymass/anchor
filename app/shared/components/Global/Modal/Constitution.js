// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Button, Header, Modal } from 'semantic-ui-react';

class ModalConstitution extends Component<Props> {
  state = { open: false }

  componentWillReceiveProps(nextProps) {
    const { isUser, settings } = nextProps;
    if (
      isUser
      && !settings.acceptedConstitution
    ) {
      this.setState({ open: true });
    }
  }

  accept = () => {
    const { actions } = this.props;
    this.setState({ open: false }, () => {
      actions.setSetting('acceptedConstitution', true);
    });
  }

  render() {
    const {
      isUser,
      settings,
      t
    } = this.props;
    const {
      open
    } = this.state;
    return (
      <Modal
        closeOnDimmerClick={false}
        closeOnDocumentClick={false}
        open={open}
      >
        <Modal.Header>
          {t('constitution_title')}
        </Modal.Header>
        <Modal.Content>
          <Header>
            {t('constitution_preface')}
          </Header>
          <Header size="small">{t('constitution_article_01_title')}</Header>
          <p>{t('constitution_article_01_body')}</p>
          <Header size="small">{t('constitution_article_02_title')}</Header>
          <p>{t('constitution_article_02_body')}</p>
          <Header size="small">{t('constitution_article_03_title')}</Header>
          <p>{t('constitution_article_03_body')}</p>
          <Header size="small">{t('constitution_article_04_title')}</Header>
          <p>{t('constitution_article_04_body')}</p>
          <Header size="small">{t('constitution_article_05_title')}</Header>
          <p>{t('constitution_article_05_body')}</p>
          <Header size="small">{t('constitution_article_06_title')}</Header>
          <p>{t('constitution_article_06_body')}</p>
          <Header size="small">{t('constitution_article_07_title')}</Header>
          <p>{t('constitution_article_07_body')}</p>
          <Header size="small">{t('constitution_article_08_title')}</Header>
          <p>{t('constitution_article_08_body')}</p>
          <Header size="small">{t('constitution_article_09_title')}</Header>
          <p>{t('constitution_article_09_body')}</p>
          <Header size="small">{t('constitution_article_10_title')}</Header>
          <p>{t('constitution_article_10_body')}</p>
          <Header size="small">{t('constitution_article_11_title')}</Header>
          <p>{t('constitution_article_11_body')}</p>
          <Header size="small">{t('constitution_article_12_title')}</Header>
          <p>{t('constitution_article_12_body')}</p>
          <Header size="small">{t('constitution_article_13_title')}</Header>
          <p>{t('constitution_article_13_body')}</p>
          <Header size="small">{t('constitution_article_14_title')}</Header>
          <p>{t('constitution_article_14_body')}</p>
          <Header size="small">{t('constitution_article_15_title')}</Header>
          <p>{t('constitution_article_15_body')}</p>
          <Header size="small">{t('constitution_article_16_title')}</Header>
          <p>{t('constitution_article_16_body')}</p>
          <Header size="small">{t('constitution_article_17_title')}</Header>
          <p>{t('constitution_article_17_body')}</p>
          <Header size="small">{t('constitution_article_18_title')}</Header>
          <p>{t('constitution_article_18_body')}</p>
          <Header size="small">{t('constitution_article_19_title')}</Header>
          <p>{t('constitution_article_19_body')}</p>
          <Header size="small">{t('constitution_article_20_title')}</Header>
          <p>{t('constitution_article_20_body')}</p>
        </Modal.Content>
        <Modal.Actions>
          <Button
            content={t('constitution_accept')}
            icon="external"
            labelPosition="right"
            onClick={this.accept}
            positive
          />
        </Modal.Actions>
      </Modal>
    );
  }
}

export default translate('constitution')(ModalConstitution);
