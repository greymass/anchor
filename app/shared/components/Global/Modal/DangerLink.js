// @flow
import React, { Component } from 'react';
import { I18n } from 'react-i18next';
import { Button, Header, Modal } from 'semantic-ui-react';
import debounce from 'lodash/debounce';

const { shell } = require('electron');

export default class ModalDangerLink extends Component<Props> {
  state = { open: false }

  isSafeish = (url) => url.startsWith('http:') || url.startsWith('https:')

  openLink() {
    const { link } = this.props;
    if (this.isSafeish(link)) {
      shell.openExternal(link);
    }
    this.setState({
      open: false
    });
  }

  openWarning() {
    this.setState({ open: true });
  }

  close = () => this.setState({ open: false })

  render() {
    const {
      open
    } = this.state;
    const {
      content,
      link
    } = this.props;
    let el = content || link;
    if (this.isSafeish(link)) {
      el = (
        <a
          onClick={this.openWarning.bind(this)}
          onKeyPress={this.openWarning.bind(this)}
          role="link"
          style={{ cursor: 'pointer' }}
          tabIndex={0}
        >
          {content || link}
        </a>
      );
    }
    return (
      <I18n ns="global">
        {
          (t) => (
            <div>
              {el}
              <Modal size="tiny" open={open} onClose={this.close}>
                <Modal.Header>
                  {t('global_dangerlink_warning_title')}
                </Modal.Header>
                <Modal.Content>
                  <p>
                    {t('global_dangerlink_warning_body_1')}
                  </p>
                  <Header>
                    {t('global_dangerlink_warning_body_2')}
                  </Header>
                  <p>
                    {t('global_dangerlink_warning_body_3')}
                  </p>
                </Modal.Content>
                <Modal.Actions>
                  <Button
                    content={t('cancel')}
                    onClick={this.close}
                    negative
                  />
                  <Button
                    positive
                    onClick={this.openLink.bind(this)}
                    icon="external"
                    labelPosition="right"
                    content={t('confirm')}
                  />
                </Modal.Actions>
              </Modal>
            </div>
          )
        }
      </I18n>
    );
  }
}
