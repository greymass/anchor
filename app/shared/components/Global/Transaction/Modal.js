// @flow
import React, { Component } from 'react';
import { Button, Header, Modal } from 'semantic-ui-react';

import GlobalTransactionHandler from './Handler';

type Props = {
  actionName: string,
  actions: {
    clearSystemState: () => void
  },
  button: {},
  content: {},
  icon: string,
  title: string,
  settings: {},
  system: {}
};

class GlobalTransactionModal extends Component<Props> {
  props: Props;

  constructor(props) {
    super(props);
    this.state = {
      open: (props.open) ? props.open : false
    };
  }

  handleOpen = () => this.setState({ open: true });

  onClose = () => this.setState({ open: false }, () => {
    this.props.actions.clearSystemState();
  });

  onOpen = () => {
    const { onOpen } = this.props;
    if (onOpen) {
      onOpen();
    }
  }

  render() {
    const {
      actionName,
      button,
      content,
      icon,
      title,
      settings,
      system
    } = this.props;
    let {
      transaction
    } = this.props;
    const {
      open
    } = this.state;
    // Load the transaction from props by default, but overwrite
    //   with last transaction from the system if exists
    if (system && system[`${actionName}_LAST_TRANSACTION`]) {
      transaction = system[`${actionName}_LAST_TRANSACTION`];
    }
    return (
      <Modal
        centered={false}
        closeIcon
        closeOnDimmerClick={false}
        closeOnDocumentClick={false}
        trigger={(
          <Button
            color={button.color}
            content={button.content}
            fluid
            icon={button.icon}
            onClick={this.handleOpen}
          />
        )}
        open={open}
        onOpen={this.onOpen}
        onClose={this.onClose}
      >
        <Header
          icon={icon}
          content={title}
        />
        <Modal.Content>
          <GlobalTransactionHandler
            actionName={actionName}
            content={content}
            onClose={this.onClose}
            settings={settings}
            system={system}
            transaction={transaction}
          />
        </Modal.Content>
      </Modal>
    );
  }
}

export default GlobalTransactionModal;
