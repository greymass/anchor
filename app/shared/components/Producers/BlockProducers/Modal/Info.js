// @flow
import React, { Component } from 'react';
import { withTranslation } from 'react-i18next';
import { Modal, Tab } from 'semantic-ui-react';

import ProducersModalInfoDetails from './Info/Details';
import ProducersModalInfoData from './Info/Data';

class ProducersModalInfo extends Component<Props> {
  render() {
    const {
      onClose,
      producerInfo,
      settings,
      t,
      viewing
    } = this.props;
    const panes = [
      { menuItem: t('producer_info_menu_item_one'), render: () => <Tab.Pane><ProducersModalInfoDetails settings={settings} producerInfo={producerInfo} /></Tab.Pane> },
      { menuItem: t('producer_info_menu_item_two'), render: () => <Tab.Pane><ProducersModalInfoData producerInfo={producerInfo} /></Tab.Pane> },
    ];
    return (
      <Modal
        closeIcon
        onClose={onClose}
        open={!!(viewing)}
      >
        <Modal.Header>
          {t('producers_info_header', { producer: viewing })}
        </Modal.Header>
        <Modal.Content>
          <Tab panes={panes} />
        </Modal.Content>
      </Modal>
    );
  }
}

export default withTranslation('producers')(ProducersModalInfo);
