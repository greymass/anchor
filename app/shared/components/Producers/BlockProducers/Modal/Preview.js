// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';

import GlobalTransactionModal from '../../../Global/Transaction/Modal';
import ProducersVotingPreviewSelection from './Preview/Selection';

class ProducersVotingPreview extends Component<Props> {
  render() {
    const {
      actions,
      blockExplorers,
      lastError,
      onConfirm,
      selected,
      settings,
      submitting,
      system,
      t
    } = this.props;

    return (
      <GlobalTransactionModal
        actionName="VOTEPRODUCER"
        actions={actions}
        blockExplorers={blockExplorers}
        button={{
          color: 'green',
          content: t('producer_voter_save_changes'),
          fluid: true,
          icon: ''
        }}
        content={(
          <ProducersVotingPreviewSelection
            lastError={lastError}
            onConfirm={onConfirm}
            selected={selected}
            settings={settings}
            submitting={submitting}
          />
        )}
        icon="exchange"
        title={t('producer_voter_save_changes')}
        settings={settings}
        system={system}
      />
    );
  }
}

export default translate('producers')(ProducersVotingPreview);
