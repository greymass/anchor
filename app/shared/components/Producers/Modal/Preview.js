// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';

import GlobalTransactionModal from '../../Global/Transaction/Modal';
import ProducersVotingPreviewSelection from './Preview/Selection';

class ProducersVotingPreview extends Component<Props> {
  render() {
    const {
      actions,
      lastError,
      onConfirm,
      selected,
      settings,
      submitting,
      system,
      t
    } = this.props;

    return (
      <GlobalTransactionModal
        actionName="VOTEPRODUCER"
        actions={actions}
        button={{
          color: 'green',
          content: t('producer_voter_save_changes'),
          icon: ''
        }}
        content={(
          <ProducersVotingPreviewSelection
            lastError={lastError}
            onConfirm={onConfirm}
            selected={selected}
            submitting={submitting}
          />
        )}
        icon="exchange"
        title={t('producer_voter_save_changes')}
        settings={settings}
        system={system}
      />
    );
  }
}

export default translate('producers')(ProducersVotingPreview);
