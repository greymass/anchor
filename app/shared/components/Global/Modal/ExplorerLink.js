// @flow
import React, { Component } from 'react';
import { isEmpty } from 'lodash';
import DangerLink from './DangerLink';

export default class GlobalModalExplorerLink extends Component<Props> {
  render() {
    const {
      blockExplorers,
      content,
      linkData,
      linkType,
      settings
    } = this.props;

    let {
      linkBlockId
    } = this.props;

    const { blockExplorer: selected } = settings;

    if (!blockExplorers || isEmpty(blockExplorers)) return content;

    const blockExplorer = (selected in Object.keys(blockExplorers))
      ? blockExplorers[selected]
      : blockExplorers[Object.keys(blockExplorers)[0]];

    if (!blockExplorer[linkType]) {
      return false;
    }

    if (linkData || linkType === 'account') {
      const urlPartsWithoutVariable = blockExplorer[linkType].split(`{${linkType}}`);

      let generatedLink = null;
      if ((linkType === 'txid')) {
        if (linkBlockId && (urlPartsWithoutVariable[0] === 'https://explore.beos.world/transactions/' || (urlPartsWithoutVariable[0] === 'https://explore.testnet.beos.world/transactions/'))) {
          linkBlockId = linkBlockId && linkBlockId.toString();
          generatedLink = `${urlPartsWithoutVariable[0]}${linkBlockId}/${linkData}${urlPartsWithoutVariable[1]}`;
        } else {
          generatedLink = `${urlPartsWithoutVariable[0]}${linkData}${urlPartsWithoutVariable[1]}`;
        }
      } else {
        generatedLink = `${urlPartsWithoutVariable[0]}${linkData}${urlPartsWithoutVariable[1]}`;
      }

      return (
        <DangerLink
          content={content}
          link={generatedLink}
          settings={settings}
        />
      );
    }

    return content;
  }
}
