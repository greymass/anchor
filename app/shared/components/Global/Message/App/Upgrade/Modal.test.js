import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalMessageAppUpgradeModal } from './Modal';

const mockProps = {
  actions: {},
  constants: {},
  name: 'Modal Name',
  version: 1,
};

describe('GlobalFragmentWalletTypes', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalMessageAppUpgradeModal {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
