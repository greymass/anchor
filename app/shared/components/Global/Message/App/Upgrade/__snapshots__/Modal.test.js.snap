// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFragmentWalletTypes renders correctly 1`] = `
<Modal
  centered={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  content={
    <Translate(GlobalMessageAppUpgrade)
      actions={Object {}}
      constants={Object {}}
      name="Modal Name"
      version={1}
    />
  }
  dimmer={true}
  eventPool="Modal"
  open={true}
  size="small"
/>
`;
